### PR TITLE
[SIM_FLUTTER-31] [BE] Create User Login API

### PR DESCRIPTION
**Pull request by jstephend-sun**
### Issue Links
- [backlog](https://framgiaph.backlog.com/view/SIM_FLUTTER-31)
- [slack](https://sun-philippine.slack.com/archives/C059L7XAXFB/p1686887255953799)
- dependent #9 

###  Notes
- refresh the database since there was a wrong setup from the migrations, `php artisan migrate:fresh`
- seed your database with users `php artisan db:seed --class=UserSeeder` or together with db refresh `php artisan migrate:fresh --seder=UserSeeder`
- login at [login postman link](https://sph-flutter-simu.postman.co/workspace/SPH_FLUTTER~166b50f1-a8e9-4146-bf86-c471bcffa801/request/21124581-81b59929-06e2-401f-b5d1-1be53feb5dc4?ctx=documentation) , may have to accept [invite ](https://app.getpostman.com/join-team?invite_code=39372f93b32b82ba5a505c8f49c1f0b9)first to the workspace using comp email 
- email from one of your generated users, password is "password"

image

- check if you are logged in at [postman](https://sph-flutter-simu.postman.co/workspace/SPH_FLUTTER~166b50f1-a8e9-4146-bf86-c471bcffa801/request/21124581-bb90afb4-5b5f-480c-b776-b89df845cd0e?ctx=documentation)
- in `Authorization` tab, select `Bearer Token` and input the token you received from logging in

image

### Definition of done
- [x] create login api
- [x] create guarded route with auth

### Recording
- 1

image

- 2

image

### CODE REVIEW

**Review by Dbulawan13 — DISMISSED** [review]
Approve PR #10 

Working:
- Able to create API for user to login
- Able to login using postman
- Once a user logs in, a bearer token will be generated, and I am able to verify that the user is logged in.




Testing hours: 0.16
Comment: 0
Iteration: 0

File: backend/app/Exceptions/Handler.php [hunk 1]
```diff
@@ -34,8 +36,21 @@ class Handler extends ExceptionHandler
      */
     public function register()
     {
-        $this->reportable(function (Throwable $e) {
-            //
+        // $this->reportable(function (Throwable $e) {
+        // });
```
Comment (rdelrosariojr) [inline backend/app/Exceptions/Handler.php:40]: please remove if not used

File: backend/database/factories/UserFactory.php [hunk 2]
```diff
@@ -15,11 +15,19 @@ class UserFactory extends Factory
...
+            'first_name' => $this->faker->firstName(),
+            'last_name' => $this->faker->lastName(),
+            'middle_name' => $this->faker->lastName(),
             'email' => $this->faker->unique()->safeEmail(),
-            'email_verified_at' => now(),
-            'password' => '$2y$10$92IXUNpkjO0rOQ5byMi.Ye4oKoEa3Ro9llC/.og/at2.uheWG/igi', // password
-            'remember_token' => Str::random(10),
+            'address' => $this->faker->unique()->address(),
+            'birthday' => $this->faker->dateTime(),
+            // 'remember_token' => Str::random(10),
```
Comment (rdelrosariojr) [inline backend/database/factories/UserFactory.php:30]: remove if not needed

```suggestion
            // 'remember_token' => Str::random(10),
```

File: backend/database/factories/UserFactory.php [hunk 3]
```diff
@@ -15,11 +15,19 @@ class UserFactory extends Factory
     public function definition()
     {
         return [
-            'name' => $this->faker->name(),
+            'id' => $this->faker->unique()->uuid(),
+            'user_name' => $this->faker->userName(),
+            // password attribute is mutated when being created, see backend\app\Models\User.php
+            // 'password' => '$2y$10$92IXUNpkjO0rOQ5byMi.Ye4oKoEa3Ro9llC/.og/at2.uheWG/igi', // password <-- this is default from app creation
```
Comment (rdelrosariojr) [inline backend/database/factories/UserFactory.php:21]: we can remove the since it was already explained in the model

**Review by rdelrosariojr — COMMENTED** [review]
Reviewed time: 0.2
comments: 3

File: backend/app/Exceptions/Handler.php [hunk 4]
```diff
@@ -34,8 +36,21 @@ class Handler extends ExceptionHandler
      */
     public function register()
     {
-        $this->reportable(function (Throwable $e) {
-            //
+        // $this->reportable(function (Throwable $e) {
```
Comment (rdelrosariojr) [inline backend/app/Exceptions/Handler.php]: please remove if not needed

```suggestion
       
```

File: backend/app/Http/Controllers/UserController.php [hunk 5]
```diff
@@ -2,13 +2,38 @@
...
+    public function login(Request $request)
+    {
+        $credentials = $this->validate($request, [
+            'email' => ['required', Rule::exists('users', 'email')],
+            'password' => ['required'],
+        ]);
+
+        if (! Auth::attempt($credentials)) {
+            throw ValidationException::withMessages([
+                'email' => 'Your provided credentials could not be verified.',
```
Comment (rdelrosariojr) [inline backend/app/Http/Controllers/UserController.php:29]: Just a question, when will this appear? I tried logging in with wrong credentials but still did not appear.

**Review by rdelrosariojr — COMMENTED** [review]
reviewed time: 0.20
comments: 2

**Review by rdelrosariojr — APPROVED** [review]
APPROVED PR10

reviewed time: 0.21
comments: 1

**Pull request merged**